### PR TITLE
Fix multiple issues with modules.pkgng.latest_version

### LINKED ISSUE #60105: [BUG] `IndexError: pop from empty list` when pkgng.latest_version queries nonexistent package
**Description**

**Setup**
The `pkgng.latest_version` function will raise an exception if you search for a package that is not present in the package server.

**Steps to Reproduce the behavior**
```
$ sudo salt-call pkg.latest_version doesnotexist
[ERROR   ] Command 'pkg' failed with return code: 1
[ERROR   ] retcode: 1
[ERROR   ] Command 'pkg' failed with return code: 1
[ERROR   ] output:
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
IndexError: pop from empty list
Traceback (most recent call last):
  File "/usr/local/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3002.6', 'console_scripts', 'salt-call')()
  File "/usr/local/lib/python3.7/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/local/lib/python3.7/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/modules/pkgng.py", line 345, in latest_version
    ).pop(0),
IndexError: pop from empty list
Traceback (most recent call last):
  File "/usr/local/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3002.6', 'console_scripts', 'salt-call')()
  File "/usr/local/lib/python3.7/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/local/lib/python3.7/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/local/lib/python3.7/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/salt/modules/pkgng.py", line 345, in latest_version
    ).pop(0),
```

**Expected behavior**
It should've returned something like
```
local:
```

**Versions Report**
```
$ salt -
-versions-report
Salt Version:
          Salt: 3002.6
 
Dependency Versions:
          cffi: 1.14.5
      cherrypy: Not Installed
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.1.11
        Jinja2: 2.11.2
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.20
      pycrypto: Not Installed
  pycryptodome: 3.10.1
        pygit2: Not Installed
        Python: 3.7.10 (default, Mar 10 2021, 03:22:47)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 22.0.3
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.1
 
System Versions:
          dist: freebsd 13.0 
        locale: UTF-8
       machine: amd64
       release: 13.0-AXCIENT2
        system: FreeBSD
       version: FreeBSD 13.0 

```
</details>

**Additional context**
Patch in progress...

### LINKED ISSUE #60108: [BUG] pkgng.latest_version does not work if an origin is specified
**Description**
FreeBSD packages can be specified by their name (`bash`) or by their origin (`shells/bash`).  The `pkg` tool understands both.  Salt's pkgng.latest_version function attempts to support origins (see 51fb2ac4c7a53240723c4922acac8fed9a130348) but fails.  For one thing, the `pkg` invocation may return many results since it searches for any string match in packages' names.  For another, Salt's `_get_version` function will filter out the correct result because `pkg` returns a list of packages, not a list of origins.

**Steps to Reproduce the behavior**
```
$ sudo salt-call pkg.latest_version misc/cpuid
local:
```
Note that the pkg command executed is `pkg search -q -U misc/cpuid`.

**Expected behavior**
It should've returned `3.3_7`.  And the command executed should've been `pkg search -S origin -q -e -U misc/cpuid`

**Versions Report**
```
$ salt --versions-report
Salt Version:
          Salt: 3002.6
 
Dependency Versions:
          cffi: 1.14.5
      cherrypy: Not Installed
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.1.11
        Jinja2: 2.11.2
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.20
      pycrypto: Not Installed
  pycryptodome: 3.10.1
        pygit2: Not Installed
        Python: 3.7.10 (default, Mar 10 2021, 03:22:47)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 22.0.3
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.1
 
System Versions:
          dist: freebsd 13.0 
        locale: UTF-8
       machine: amd64
       release: 13.0-AXCIENT2
        system: FreeBSD
       version: FreeBSD 13.0 
```

### PR DESCRIPTION
**Pull request by asomers**
### What does this PR do?
Cleans up modules.pkg.latest_version, adds tests, and fixes two bugs.

### What issues does this PR fix or reference?
Fixes: #60105
Fixes: #60108

### Previous Behavior
* 60105: pkgng.latest_version raises IndexError if the package isn't found in the repository
* 60108: pkgng.latest_version fails when the package is specified by origin

### New Behavior
* 60105: When the package isn't found, pkgng.latest_version returns "", as expected
* 60108: pkgng.latest_version works when the package is specified by origin (e.g. "shells/bash")

### Merge requirements satisfied?
- [N/A] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/pytests/unit/modules/test_pkgng.py [hunk 1]
```diff
@@ -19,6 +19,80 @@ def pkgs():
     ]
 
 
+def test_latest_version(pkgs):
+    "Test basic usage of pkgng.latest_version"
```
Comment (krionbsd) [inline tests/pytests/unit/modules/test_pkgng.py:23]: Please use docstrings

**Review by krionbsd — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_pkgng.py (hunk 1 above)
Comment (krionbsd) [reply tests/pytests/unit/modules/test_pkgng.py:23]: Please fix pre-commit as well

**Comment by asomers** [comment]
@krionbsd style issues fixed

**Review by krionbsd — APPROVED** [review]

File: salt/modules/pkgng.py [hunk 2]
```diff
@@ -296,52 +296,30 @@ def latest_version(*names, **kwargs):
     chroot = kwargs.get("chroot")
     refresh = kwargs.get("refresh")
     root = kwargs.get("root")
-    pkgs = list_pkgs(versions_as_list=True, jail=jail, chroot=chroot, root=root)
-
-    if salt.utils.versions.compare(
-        _get_pkgng_version(jail, chroot, root), ">=", "1.6.0"
-    ):
-        quiet = True
```
Comment (Ch3LL) [inline salt/modules/pkgng.py:304]: Why are you removing quiet?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/pkgng.py (hunk 2 above)
Comment (asomers) [reply salt/modules/pkgng.py:304]: If you look below, you'll see that I made the `-q` flag unconditional.  Previously, the only reason not to use it was to support pkg older than 1.6.0.  My commit 47fb63d drops support for that old pkg version.

Comment (Ch3LL) [reply salt/modules/pkgng.py:304]: It looks like in a commit after that one you ended up removing `cmd.append("-q")` all together though. If I'm understanding correctly newer versions of pkg would need the `-q` flag, so shouldn't we keep it if we are only supporting newer versions?

Comment (Ch3LL) [reply salt/modules/pkgng.py:304]: K I see it now. I was grepping for `-q`. You included it in `"-eqS"`. Thanks for helping my eyesight ;)

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**